modules/airgap/setup/get_artifacts.sh

```diff
@@ -39,21 +39,29 @@ get_assets() {
   elif [[ "$product" == "rke2" ]]; then
     url="https://github.com/rancher/rke2/releases/download/$version"
     wget "$url"/sha256sum-"$arch".txt
-    wget "$url"/rke2-images.linux-"$arch".txt
+    # Ref: https://docs.rke2.io/install/airgap
+    if [[ -n "$flags" ]] && [[ "$flags" =~ "cni" ]]; then
+      wget "$url"/rke2-images-core.linux-"$arch".txt
+      if [ -n "$tarball_type" ]; then
+        wget "$url"/rke2-images-core.linux-"$arch"."$tarball_type"
```

Comment:
since we have a bunch of wgets, can we add a "download" or something, function? 

like doing the wget -O to somewhere and handling possible error

than u just call this func whenever needed on get assets and get_cni_assets

Reply:
*design

Reply:
Yeah I was thinking along the same line like adding a retry logic in case it fails to download...

Reply:
Updated please check